src/app/core/relations/relations.js

```diff
@@ -2,141 +2,237 @@ const { base, inherit } = require('core/utils/utils');
 const G3WObject = require('core/g3wobject');
 const Relation = require('core/relations/relation');
 
-// class Relations
-function Relations(options={}) {
-  const {relations} = options;
-  //store relations
+/**
+ * Relations Class
+ * 
+ * @param options
+ * 
+ * @constructor
+ */
+function Relations(options = {}) {
+
+  /**
+   * Relations store 
+   */
   this._relations = {};
-  this._length = relations ? relations.length : 0;
-  // to build relations between layers
-  this._relationsInfo = {
-    children: {}, // array child (unique ids)
-    fathers: {}, // array father (unique ids)
-    father_child: {} // info parent child
-  };
+
+  /**
+   * Number of relations 
+   */
+  this._length = options.relations ? options.relations.length : 0;
+
   let relation;
-  relations.forEach(relationConfig => {
+  options.relations.forEach(relationConfig => {
     relation = new Relation(relationConfig);
     this._relations[relation.getId()] = relation;
   });
-  this._createRelationsInfo();
+
+  this._reloadRelationsInfo();
+
   base(this);
 }
 
 inherit(Relations, G3WObject);
 
 const proto = Relations.prototype;
 
+/**
+ * Populate `this._relationsInfo` object.
+ */
 proto._createRelationsInfo = function() {
-  let father;
-  let child;
-  Object.entries(this._relations).forEach(([relationKey, relation]) => {
-    father = relation.getFather();
-    child = relation.getChild();
-    this._relationsInfo.father_child[father+child] = relationKey;
-    if (!this._relationsInfo.fathers[father]) this._relationsInfo.fathers[father] = [];
-    if (!this._relationsInfo.children[child]) this._relationsInfo.children[child] = [];
-    this._relationsInfo.fathers[father].push(child);
-    this._relationsInfo.children[child].push(father);
+
+  // sanity check
+  if (!this._relationsInfo) {
+    this._clearRelationsInfo();
+  }
+
+  let f, c;
+  const { father_child, fathers, children } = this._relationsInfo;
+
+  Object
+    .entries(this._relations)
+    .forEach(([relationKey, relation]) => {
+
+      f = relation.getFather();
+      c = relation.getChild();
+
+      father_child[f + c] = relationKey;       // [f + c] = the relation key ?
+      fathers[f]          = fathers[f]  || [];
+      children[c]         = children[c] || [];
+
+      fathers[f].push(c);
+      children[c].push(f);
   });
 };
 
+/**
+ * @private
+ */
 proto._clearRelationsInfo = function() {
   this._relationsInfo = {
-    children: {},
-    fathers: {},
-    father_children: {}
+    children:     {},     // array child (unique ids)
+    fathers:      {},     // array father (unique ids)
+    father_child: {},     // info parent child
```

Comment:
@volterra79 double check this assignment.

it should be `father_children` or `father_child`?

Reply:
@Raruto it is correct fater_child. It store fater layerId and child layerId as attribute and relationId as value